util/patch/patch_test.go

```diff
@@ -21,16 +21,61 @@ import (
 	"reflect"
 	"testing"
 
+	. "github.com/onsi/gomega"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha2"
 )
 
+func TestHelperUnstructuredPatch(t *testing.T) {
+	RegisterTestingT(t)
+	ctx := context.TODO()
+
+	obj := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"kind":       "BootstrapConfig",
+			"apiVersion": "bootstrap.cluster.x-k8s.io/v1alpha2",
+			"metadata": map[string]interface{}{
+				"name":      "test-bootstrap",
+				"namespace": "default",
+			},
+		},
+	}
+	fakeClient := fake.NewFakeClient()
+	fakeClient.Create(ctx, obj)
+
+	h, err := NewHelper(obj, fakeClient)
+	if err != nil {
+		t.Fatalf("Expected no error initializing helper: %v", err)
+	}
+
+	refs := []metav1.OwnerReference{
+		{
+			APIVersion: "cluster.x-k8s.io/v1alpha2",
+			Kind:       "Cluster",
+			Name:       "test",
+		},
+	}
+	obj.SetOwnerReferences(refs)
+
+	err = h.Patch(ctx, obj)
+	Expect(err).ToNot(HaveOccurred())
+
+	afterObj := obj.DeepCopy()
```

Comment:
You could potentially initialize this to an empty Unstructured, right?

Reply:
The object we pass to client-go would still need Kind/APIVersion etc. If I move to an empty unstructured, it'll return
```
Unexpected error:
            <*runtime.missingKindErr | 0xc000108fe0>: {
                data: "unstructured object has no kind",
            }
            Object 'Kind' is missing in 'unstructured object has no kind'
        occurred
```